src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -202,7 +197,35 @@ protected static Model.Identifier GenerateObservationIdentifier(IObservationGrou
         protected virtual async Task<Model.Observation> GetObservationFromServerAsync(Model.Identifier identifier)
         {
             var result = await _fhirService.SearchForResourceAsync(Model.ResourceType.Observation, identifier.ToSearchQueryParameter()).ConfigureAwait(false);
-            return await result.ReadOneFromBundleWithContinuationAsync<Model.Observation>(_fhirService);
+
+            var foundObservations = (await result.ReadFromBundleWithContinuationAsync<Model.Observation>(_fhirService).ConfigureAwait(false))
+                .ToArray();
+
+            if (foundObservations.Length == 0)
+            {
+                return null;
+            }
+
+            // Discovered an issue where FHIR Service is only matching on first 128 characters of the identifier.
+            // Match observations that match the desired identifier.
+            // This will also prevent merging observations that share the first 128 characters.
+
+            var matchedObservations = foundObservations

```

Comment:
nit: Should we log (warning) the fact that we even got multiple `foundObservations` for a given identifier (even post the fhir service bug, since it is not an expected scenario)?